src/main/java/org/opentestsystem/ap/ims/entity/SharedSavedSearch.java

```diff
@@ -0,0 +1,59 @@
+package org.opentestsystem.ap.ims.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.persistence.Embeddable;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinColumns;
+import javax.persistence.ManyToOne;
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+
+@Getter
+@Entity
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode
+public class SharedSavedSearch implements Serializable {
+
+    @Data
+    @Embeddable
+    @NoArgsConstructor
+    @AllArgsConstructor
+    @EqualsAndHashCode
+    @ToString
+    public static class SharedSavedSearchIdentity implements Serializable {
+
+        @NotNull
+        private String ownerUserName;
+
+        @NotNull
+        private String searchName;
+
+        @NotNull
+        private String sharedWithUserName;
+    }
+
+    @EmbeddedId
+    private SharedSavedSearchIdentity id;
+
+    @ManyToOne
+    @JoinColumns({
+        @JoinColumn(name = "ownerUserName", referencedColumnName = "userName", insertable = false, updatable = false),
+        @JoinColumn(name = "searchName", referencedColumnName = "searchName", insertable = false, updatable = false)
+    })
+    private SavedSearch savedSearch;
```

Comment:
This mapping was necessary so we can query to get all the saved searches shared with a user.